lib/matplotlib/axes/_axes.py

```diff
@@ -6638,22 +6657,38 @@ def hist(self, x, bins=None, range=None, density=None, weights=None,
                     "sets and %d colors were provided" % (nx, len(color)))
                 raise ValueError(error_message)
 
-        # If bins are not specified either explicitly or via range,
-        # we need to figure out the range required for all datasets,
-        # and supply that to np.histogram.
-        if not binsgiven and not input_empty:
+        hist_kwargs = dict()
+
+        # if the bin_range is not given, compute without nan numpy
+        # does not do this for us when guessing the range (but will
+        # happily ignore nans when computing the histogram).
+        if bin_range is None:
             xmin = np.inf
             xmax = -np.inf
             for xi in x:
-                if len(xi) > 0:
+                if len(xi):
                     xmin = min(xmin, np.nanmin(xi))
                     xmax = max(xmax, np.nanmax(xi))
-            bin_range = (xmin, xmax)
+            if xmax > xmin:
```

Comment:
looks like this line is only to handle the case where all x's are the same?  but is that really needed? (why was it not there before?)

Reply:
was thinking about the all-nan case, should probably be `>=`?

Reply:
pushed a change for this